src/lib/device-types/index.ts

```diff
@@ -103,29 +107,44 @@ function fetchDeviceTypes(): Promise<Dictionary<DeviceTypeInfo>> {
 				}
 
 				const sortedBuilds = sortBuildIds(builds!);
-				return getFirstValidBuild(slug, sortedBuilds).then(latestBuildInfo => {
-					if (!latestBuildInfo) {
-						return;
-					}
-
-					result[slug] = {
-						versions: builds,
-						latest: latestBuildInfo,
-					};
-
-					_.forEach(
-						(latestBuildInfo.deviceType as DeviceTypeWithAliases).aliases,
-						alias => {
-							result[alias] = result[slug];
-						},
-					);
-				});
+				return getFirstValidBuild(slug, sortedBuilds)
+					.then(latestBuildInfo => {
+						if (!latestBuildInfo) {
+							return;
+						}
+
+						result[slug] = {
+							versions: builds,
+							latest: latestBuildInfo,
+						};
+
+						_.forEach(
+							(latestBuildInfo.deviceType as DeviceTypeWithAliases).aliases,
+							alias => {
+								result[alias] = result[slug];
+							},
+						);
+					})
+					.tapCatch(err => {
+						captureException(
+							err,
+							`Failed to find valid build data for device type: ${slug}`,
+						);
+					});
 			});
 		})
 		.return(result)
 		.catch(err => {
-			captureException(err, 'Failed to get device types');
-			return Promise.delay(RETRY_DELAY).then(fetchDeviceTypes);
+			captureException(
+				{ ...err, retryAttemptsLeft },
+				'Failed to get device types',
+			);
+			if (retryAttemptsLeft >= 1) {
+				return Promise.delay(RETRY_DELAY).then(() =>
+					fetchDeviceTypes(retryAttemptsLeft - 1),
+				);
+			}
+			throw new InternalRequestError(err);
```

Comment:
This ensures that we will not return any details about the error in the response body both when used in a route, as well as when used in a hook.